public/src/components/channelManagement/testEditor.tsx

```diff
@@ -41,279 +40,240 @@ interface DataFromServer<T extends Test> {
   userEmail: string;
 }
 
-interface TestFormState<T extends Test> {
-  tests: T[] | null;
-  version: string | null;
-  selectedTestName?: string;
-  editMode: boolean;
-  lockStatus: LockStatus;
-  modifiedTests: ModifiedTests;
-  timeoutAlertId: number | null; // A timeout for warning about being open for edit for too long
-}
+const useEditModeAlertTimer = (editMode: boolean): void => {
+  const [timeoutId, setTimeoutId] = useState<number | undefined>(undefined);
+
+  const clearTimeout = (): void => {
+    window.clearTimeout(timeoutId);
+    setTimeoutId(undefined);
+  };
+
+  useEffect(() => {
+    if (editMode) {
+      if (timeoutId) {
+        clearTimeout();
+      }
+      const newTimeoutId = window.setTimeout(() => {
+        alert(
+          "You've had this editing session open for 20 minutes - if you leave it much longer then you may lose any unsaved work!\nIf you've finished then please either save or cancel.",
+        );
+        clearTimeout();
+      }, 60 * 20 * 1000);
+
+      setTimeoutId(newTimeoutId);
+    } else if (timeoutId) {
+      clearTimeout();
+    }
+  }, [editMode]);
+};
 
 /**
  * A stateful higher-order component for fetching/saving test data.
  * Takes care of locks
  */
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 const TestEditor = <T extends Test>(
   InnerComponent: React.ComponentType<InnerComponentProps<T>>,
   settingsType: FrontendSettingsType,
-) =>
-  class extends React.Component<{}, TestFormState<T>> {
-    state: TestFormState<T> = {
-      tests: null,
-      version: null,
-      selectedTestName: undefined,
-      editMode: false,
-      lockStatus: { locked: false },
-      modifiedTests: {},
-      timeoutAlertId: null,
-    };
+): React.FC => {
+  const WrappedInnerComponent: React.FC = () => {
+    const [tests, setTests] = useState<T[] | null>(null);
+    const [version, setVersion] = useState<string | null>(null);
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const [selectedTestName, setSelectedTestName] = useState<string>('');
```

Comment:
would `string | null` be more consistent?

Reply:
Yeah makes sense - I'll do that now 👍